src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -12,21 +12,41 @@
 using CilStrip.Mono.Cecil.Binary;
 using CilStrip.Mono.Cecil.Cil;
 using CilStrip.Mono.Cecil.Metadata;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+using System.Buffers;
 
 public class ILStrip : Microsoft.Build.Utilities.Task
 {
+    [Required]
     /// <summary>
     /// Assemblies to be stripped.
     /// The assemblies will be modified in place if OutputPath metadata is not set.
     /// </summary>
-    [Required]
     public ITaskItem[] Assemblies { get; set; } = Array.Empty<ITaskItem>();
 
     /// <summary>
     /// Disable parallel stripping
     /// </summary>
     public bool DisableParallelStripping { get; set; }
 
+    /// <summary>
+    /// Enable the feature of trimming indiviual methods
+    /// </summary>
+    public bool TrimIndividualMethods { get; set; }
+
+    /// <summary>
+    /// Assembilies got trimmed successfully.
+    ///
+    /// Successful trimming will set the following metadata on the items:
+    ///  - TrimmedAssemblyFileName
+    /// </summary>
+    [Output]
+    public ITaskItem[]? TrimmedAssemblies { get; set; }
+
+    private List<ITaskItem> trimmedAssemblies = new List<ITaskItem>();
+
     public override bool Execute()
```

Comment:
I suggest adding the pattern of using `LogAsErrorException` like https://github.com/dotnet/runtime/blob/f6868c12de1e9c378ac1eaf4477bbc14171b1c6b/src/tasks/AotCompilerTask/MonoAOTCompiler.cs#L443-L460 
.. it should make the code flow easier to follow.

Reply:
It is not clear to me how adding a try-catch would make a method easier to read. Could you elaborate more?

Reply:
I meant:
1. rename `Execute` to `ExecuteInternal`
2. Add `Execute` like https://github.com/dotnet/runtime/blob/f6868c12de1e9c378ac1eaf4477bbc14171b1c6b/src/tasks/AotCompilerTask/MonoAOTCompiler.cs#L443-L460
3. Now, instead of having to pipe `false` from inner methods back to `Execute` to fail the task, you will be able to just throw `LogAsErrorException`.

this can be done in a follow up PR. I'm thinking of adding this to the tasks that don't have it, so I can take care of it too.